include/model.php

```diff
@@ -748,53 +763,86 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
+		$new_taxonomies   = array_diff( $registered_taxonomies, $known_taxonomies );
+
+		if ( empty( $new_taxonomies ) ) {
+			// No new 3rd party language taxonomies.
+			return;
+		}
+
+		// We have at least one unknown 3rd party language taxonomy.
+		$this->add_missing_secondary_language_terms( $new_taxonomies );
+
+		// Keep the previous values, so this is triggered only once per taxonomy.
+		$this->options['3rd_party_taxonomies'] = array_merge( $known_taxonomies, $new_taxonomies );
+		update_option( 'polylang', $this->options );
 	}
 
 	/**
-	 * Returns the default language.
+	 * Adds the missing language terms for the secondary language taxonomies.
 	 *
 	 * @since 3.4
 	 *
-	 * @return PLL_Language|false Default language object, `false` if no language found.
+	 * @param string[]|null $taxonomies Optional. List of language taxonomies to deal with. An empty value means all of
+	 *                                  them. Defauls to all taxonomies.
+	 * @return void
+	 *
+	 * @phpstan-param array<non-empty-string>|null $taxonomies
 	 */
-	public function get_default_language() {
-		if ( empty( $this->options['default_lang'] ) ) {
-			return false;
+	public function add_missing_secondary_language_terms( array $taxonomies = null ) {
+		foreach ( $this->get_languages_list() as $language ) {
+			$this->update_secondary_language_terms( $language->slug, $language->name, $language, $taxonomies );
 		}
 
-		return $this->get_language( $this->options['default_lang'] );
+		// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
+		$this->clean_languages_cache();
 	}
 
 	/**
 	 * Updates or adds new terms for a secondary language taxonomy (aka not 'language').
 	 *
 	 * @since 3.4
 	 *
-	 * @param string            $slug     Language term slug (with or without the `pll_` prefix).
-	 * @param string            $name     Language name (label).
-	 * @param PLL_Language|null $language Optional. A language object. Required to update the existing terms.
+	 * @param string            $slug       Language term slug (with or without the `pll_` prefix).
+	 * @param string            $name       Language name (label).
+	 * @param PLL_Language|null $language   Optional. A language object. Required to update the existing terms.
+	 * @param string[]|null     $taxonomies Optional. List of language taxonomies to deal with. An empty value means
+	 *                                      all of them. Defauls to all taxonomies.
 	 * @return void
 	 *
 	 * @phpstan-param non-empty-string $slug
 	 * @phpstan-param non-empty-string $name
+	 * @phpstan-param array<non-empty-string>|null $taxonomies
 	 */
-	protected function update_secondary_language_terms( $slug, $name, PLL_Language $language = null ) {
+	protected function update_secondary_language_terms( $slug, $name, PLL_Language $language = null, array $taxonomies = null ) {
```

Comment:
It looks strange to me to strict type to `array` and then passing `null`.
```suggestion
	protected function update_secondary_language_terms( $slug, $name, PLL_Language $language = null, array $taxonomies = array() ) {
```

Reply:
Right. I think I used `isset( $taxonomies )` at some point, but thought it was not a good idea and didn't changed it for `array()`.
Done in [9a118a7](https://github.com/polylang/polylang/pull/1242/commits/9a118a77bdf7ff32ca551fae5ebcab1832f12d48).